### DISCUSSION TITLE
Translations not pulling correctly.

### DISCUSSION THREAD

**Discussion by holden** [swellstores/community#170] [Help] [answered]
Hi,

I just noticed today when I pushed the Origin theme github master to Vercel that the translations aren't pulling correctly. Everything else seems fine,, it's pulling products and pricing, etc. but the text of these aren't coming through, eg "products.slug.addToCart" instead of "Add to cart."

I checked the file `editor.json` and these all appear to be there, what am I doing wrong?  It also does the same thing locally.

What am I doing wrong?

<img width="686" alt="CleanShot 2023-03-07 at 20 58 06@2x" src="https://user-images.githubusercontent.com/48241/223538514-658a8cd3-e7df-4767-99ef-269e6cd1d75e.png">

**Comment by rodrigoddalmeida** [comment]
Hi @holden, since you are setting it up externally, you also need to add Origin's default values (found in `defaults.json`) to your store settings. The easiest way to do it is through the Admin Dashboard > Developer > Console and make a request like this:

`PUT /settings store/settings`
```json
{
  "$set": {
    "values": {
      ...contents of defaults.json
    }
  }
}
```

The reason you need to do that is, while `editor.json` does declare keys and default values, it doesn't apply them immediately - they are used when interacting with the Theme Editor.

**Reply by rodrigoddalmeida** [reply]
One thing to note: if you already have values stored in `/settings/store/settings/values`, you might want to merge them instead, by removing `$set` from the query operation.

```json
{
   // Removing $set will allow this object to merge with the current document where possible
    "values": {
      ...contents of defaults.json
    }
}
```

**Reply by holden** [reply]
Ok, I did this, the second one. And I can see these values now in `/settings/store/settings/values` but it's still not showing up either locally or when I redeploy to vercel. ;-(

<img width="729" alt="CleanShot 2023-03-07 at 21 30 40@2x" src="https://user-images.githubusercontent.com/48241/223545314-c5aa8dd0-5084-43b4-beb2-f90229cf4c23.png">

**Comment by holden** [comment]
I tried it a second time.  This time I overwrote all of the values. I definitely had an effect on other elements, like the logo, etc. everything except the language.

**Reply by rodrigoddalmeida** [reply]
In this case it might be related to how your locales are set, and if those values have been correctly localized as well. 

You can confirm if language tokens are being correctly loaded from the API by locally running the app and checking the contents of `/static/lang/*.json`, see if they match the locales and localized strings you've set in `/settings/store/settings`.

**Reply by holden** [reply]
Yes, they look fine, there's only one file `/static/lang/en-US.json` and it looks good `Add to cart` is plainly there.

We haven't done any major modifications, the only thing that I had to change to get it running was to require a lower version of node in the package.json, as it wouldn't work with 18.

`The engine "node" is incompatible with this module. Expected version "12 - 16"`

So we added:

`"node": ">=14.19.1 <17.0.0",`

We're on `v16.17.0` 

Our store is: https://cirqus.swell.store/products/team-building/ and hosted it's https://beta.cirqus.co/products/team-building/

I noticed it's logging a bunch of errors in the console related to the same issue.

```
[vue-i18n] Value of key 'products.slug.addToCart' is not a string or function !
eval @ client.js:141
client.js:141 [vue-i18n] Cannot translate the value of keypath 'products.slug.addToCart'. Use the value of keypath as default.
eval @ client.js:141
client.js:141 [vue-i18n] Value of key 'products.slug.updating' is not a string or function !
eval @ client.js:141
client.js:141 [vue-i18n] Cannot translate the value of keypath 'products.slug.updating'. Use the value of keypath as default.
```

**Comment by holden** [comment]
Ok, I don't quite know what the real issue was but I managed to fix it by cloning the repo listed under developer tools in the theme menu. Previously we were just using the github repo for origin: https://github.com/swellstores/origin-theme.

It didn't seem to make any difference except for this.

Thanks again

**Reply by holden** [reply]
`2.0.13` seems to be the version where this starts breaking. `2.0.12` works fine. But from `2.0.13` onwards these translations are breaking. This includes the version hosted on swell as well as locally and on vercel: https://cirqus--dev.swell.store/products/evenchilada-music-bingo/

**Comment by rodrigoddalmeida** [comment, marked as answer]
@holden a fix should be released just now @ v2.0.15

**Reply by holden** [reply]
Awesome!  Thanks

**Reply by holden** [reply]
Yes it seem to work perfectly now.  Thanks!
